File: MPI/mpiident.c
/* Beispiel-Programm aus dem Buch  */
/* »Cluster Computing« von         */
/* Heiko Bauke und Stephan Mertens */
/*                                 */
/* Springer 2005                   */
/* ISBN 3-540-42299-4              */


#include <stdio.h>
#include <stdlib.h>
#include "mpi.h"

int main(int argc, char *argv[]) {
  int myrank, nprocs, len;
  char name[MPI_MAX_PROCESSOR_NAME+1];

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
  MPI_Comm_rank(MPI_COMM_WORLD, &myrank);
  MPI_Get_processor_name(name, &len);
  name[len]='\0';
  printf("Prozess %d (von %d) läuft auf %s.\n", myrank, nprocs, name);
  MPI_Finalize();
  return EXIT_SUCCESS;
}

File: MPI/mpiident.f
C Beispiel-Programm aus dem Buch  
C »Cluster Computing« von         
C Heiko Bauke und Stephan Mertens 
C                                 
C Springer 2005                   
C ISBN 3-540-42299-4              


      PROGRAM MPIIDENT
      IMPLICIT NONE
      INCLUDE "mpif.h"
      INTEGER MYRANK, NPROCS, LENGTH, ERROR
      CHARACTER(LEN=MPI_MAX_PROCESSOR_NAME) NAME

      CALL MPI_INIT(ERROR)
      CALL MPI_COMM_SIZE(MPI_COMM_WORLD, NPROCS, ERROR)
      CALL MPI_COMM_RANK(MPI_COMM_WORLD, MYRANK, ERROR)
      CALL MPI_GET_PROCESSOR_NAME(NAME, LENGTH, ERROR)
      WRITE (*, *) 'Prozess ', MYRANK, ' (von ', NPROCS,
     +     ') läuft auf ', NAME(1:LENGTH)
      CALL MPI_FINALIZE(ERROR)
      END
